File: Prototype3/Backend/models/historial.py
class Historial:
    def __init__(self, id, child_id, data, hora, estat, totalHores):
        self.id = id
        self.child_id = child_id
        self.data = data  # Fecha en formato YYYY-MM-DD
        self.hora = hora  # Hora en formato HH:MM:SS
        self.estat = estat  # Estado del sueño
        self.totalHores = totalHores  # Total horas dormidas

    def to_dict(self):
        return {
            "id": self.id,
            "child_id": self.child_id,
            "data": self.data,
            "hora": self.hora,
            "estat": self.estat,
            "totalHores": self.totalHores
        }

File: Prototype3/Frontend/tapatapp_flutter/lib/core/models/historial.dart
import 'package:json_annotation/json_annotation.dart';

part 'historial.g.dart';

@JsonSerializable()
class Historial {
  final int id;
  @JsonKey(name: 'child_id')
  final int childId;
  final String data; // Fecha en formato YYYY-MM-DD
  final String hora; // Hora en formato HH:MM
  final String estat; // Estado del sueño
  @JsonKey(name: 'totalHores')
  final double totalHoras;

  Historial({
    required this.id,
    required this.childId,
    required this.data,
    required this.hora,
    required this.estat,
    required this.totalHoras,
  });

  factory Historial.fromJson(Map<String, dynamic> json) => _$HistorialFromJson(json);
  Map<String, dynamic> toJson() => _$HistorialToJson(this);
}
